utils/schema/generate.js

```diff
@@ -0,0 +1,160 @@
+function createDefinitions(resources, lookups, additionalProperties = false) {
+  const typeMappings = {
+    'Edm.String': 'string',
+    'Edm.Boolean': 'boolean',
+    'Edm.Int16': 'integer',
+    'Edm.Int32': 'integer',
+    'Edm.Int64': 'integer',
+    'Edm.Decimal': 'number',
+    'Edm.Single': 'number',
+    'Edm.Double': 'number',
+    'Edm.DateTimeOffset': 'string',
+    'Edm.Date': 'string'
+  };
+
+  // Preprocess the lookups data to create a hashmap
+  const lookupsMap = {};
+  lookups.forEach(lookup => {
+    if (!lookupsMap[lookup.lookupName]) {
+      lookupsMap[lookup.lookupName] = new Set();
+    }
+    lookupsMap[lookup.lookupName].add(lookup.lookupValue);
+  });
+
+  function getPossibleValuesForNamespacedType(lookupName) {
+    const possibleValues = lookupsMap[lookupName];
+    return possibleValues ? [...possibleValues] : [];
+  }
+
+  const definitions = {};
+
+  for (const [resourceName, resourceFields] of Object.entries(resources)) {
+    const properties = {};
+
+    resourceFields.forEach(field => {
+      const { fieldName } = field;
+      let schema = {};
+      if (field.isExpansion || field.isCollection) {
+        const itemTypeSchema = {};
+
+        if (field.isExpansion) {
+          itemTypeSchema['$ref'] = `#/definitions/${field.typeName}`;
+        } else {
+          const mappedType = typeMappings[field.type] || 'object';
+          itemTypeSchema['type'] = mappedType;
+          if (field.maxLength) itemTypeSchema['maxLength'] = field.maxLength;
+          if (field.type === 'Edm.DateTimeOffset') itemTypeSchema['format'] = 'date-time';
+          if (field.type === 'Edm.Date') itemTypeSchema['format'] = 'date';
+
+          // probably needs to be another condition?
+          if (field.type.startsWith('org.reso.metadata')) {
+            const possibleValues = getPossibleValuesForNamespacedType(field.type);
```

Comment:
This function should probably be called `getPossibleLookupValues`...

There are two cases where you could have namespaced types ... 
* Complex Type
* Enumeration

In all other cases, the type should start with `Edm.`, e.g. `Edm.String`.

With complex types, we would have additional nested properties, which could be enumerations, but could also be fields or other complex types or arrays. We can skip this case for now and we'll bring this in with DD 2.1.

For now, we should be focused on enumerated types. The easiest way to tell if something is an enumerated type is that if it doesn't begin with `Edm.`, then look in the Lookups for an item containing that type (lookups should be in a map in this case). Each lookup will have both a lookup name and value, which should be unique. You'll need to build the possible lookups for the given field by collecting all lookup values for a given lookup name.